addons/crm_partner_assign/wizard/crm_forward_to_partner.py

```diff
@@ -112,73 +94,56 @@ def action_forward(self, cr, uid, ids, context=None):
         return portal_link
 
-    def get_portal_url(self, cr, uid, ids, context=None):
-        portal_link = "%s/?db=%s" % (self.pool.get('ir.config_parameter').get_param(cr, uid, 'web.base.url'), cr.dbname)
+    def get_portal_url(self):
+        portal_link = "%s/?db=%s" % (self.env['ir.config_parameter'].get_param('web.base.url'), self.env.cr.dbname)
         return portal_link
+    forward_type = fields.Selection([
+        ('single', 'a single partner: manual selection of partner'), 
+        ('assigned', "several partners: automatic assignation, using GPS coordinates and partner's grades"), ], 
+        string='Forward selected leads to',
+        default=lambda self: self.env.context.get('forward_type') or 'single')
+    partner_id = fields.Many2one('res.partner', string='Forward Leads To')
+    assignation_lines = fields.One2many('crm.lead.assignation', 'forward_id', string='Partner Assignation')
+    body = fields.Html(string='Contents', help='Automatically sanitized HTML contents')
 
-    _columns = {
-        'forward_type': fields.selection([('single', 'a single partner: manual selection of partner'), ('assigned', "several partners: automatic assignation, using GPS coordinates and partner's grades"), ], 'Forward selected leads to'),
-        'partner_id': fields.many2one('res.partner', 'Forward Leads To'),
-        'assignation_lines': fields.one2many('crm.lead.assignation', 'forward_id', 'Partner Assignation'),
-        'body': fields.html('Contents', help='Automatically sanitized HTML contents'),
-    }
 
-    _defaults = {
-        'forward_type': lambda self, cr, uid, c: c.get('forward_type') or 'single',
-    }
-
-
-class crm_lead_assignation (osv.TransientModel):
+class crm_lead_assignation (models.TransientModel):
     _name = 'crm.lead.assignation'
-    _columns = {
-        'forward_id': fields.many2one('crm.lead.forward.to.partner', 'Partner Assignation'),
-        'lead_id': fields.many2one('crm.lead', 'Lead'),
-        'lead_location': fields.char('Lead Location', size=128),
-        'partner_assigned_id': fields.many2one('res.partner', 'Assigned Partner'),
-        'partner_location': fields.char('Partner Location', size=128),
-        'lead_link': fields.char('Lead  Single Links', size=128),
-    }
-
-    def on_change_lead_id(self, cr, uid, ids, lead_id, context=None):
-        if not context:
-            context = {}
-        if not lead_id:
-            return {'value': {'lead_location': False}}
-        lead = self.pool.get('crm.lead').browse(cr, uid, lead_id, context=context)
+    forward_id = fields.Many2one('crm.lead.forward.to.partner', string='Partner Assignation')
+    lead_id = fields.Many2one('crm.lead', string='Lead')
+    lead_location = fields.Char(string='Lead Location', size=128)
+    partner_assigned_id = fields.Many2one('res.partner', string='Assigned Partner')
+    partner_location = fields.Char(string='Partner Location', size=128)
```

Comment:
Is there a need of size attribute for partner_location and lead_location field ?